### PR TITLE
podio, edm4hep and lcio: add lib and lib64 to LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by jmcarcell**
Since spack doesn't fill anymore LD_LIBRARY_PATH these are needed for loading ROOT dictionaries

### CODE REVIEW

**Comment by vvolkl** [comment]
OK, for me, although there will be a ton of other packages (anything that builds a root dictionary) where this needs to be set as well. An alternative would be to always set the LD_LIBRARY_PATH in the `dependent_build_environment` and `dependent_run_environment` of ROOT. Not every package that depends on ROOT will build dictionaries, but those that do are certain to work like this, and don't need the maintenance effort in the recipes. ROOT-dependent packages are also unlikely to be system packages, so setting the LD_LIBRARY_PATH should be safe for emacs etc.

**Comment by jmcarcell** [comment]
For deciding which way I checked and the number of repos with ROOT dictionaries that we install doesn't seem that high and with this PR I think all the ones in spack upstream are covered. Having a look at which packages depend on ROOT there are many and also many don't build dictionaries so I think this PR is cleaner at the cost of having to make these changes (but this should only happen once) so for now I'm not going to make a PR to ROOT.

File: var/spack/repos/builtin/packages/podio/package.py [hunk 1]
```diff
@@ -119,19 +119,22 @@ def cmake_args(self):
 
     def setup_run_environment(self, env):
         env.prepend_path("PYTHONPATH", self.prefix.python)
-        env.prepend_path("LD_LIBRARY_PATH", self.spec["podio"].libs.directories[0])
+        env.prepend_path("LD_LIBRARY_PATH", self.spec["podio"].prefix.lib)
+        env.prepend_path("LD_LIBRARY_PATH", self.spec["podio"].prefix.lib64)
```
Comment (tmadlener) [inline var/spack/repos/builtin/packages/podio/package.py]: I woul have suggested to do exactly the opposite for `edm4hep` above. Using `libs.directories[0]` is the recommended way to do this AFAIK, why switch back to the duplication here?

Comment (jmcarcell) [reply var/spack/repos/builtin/packages/podio/package.py]: Fixed

**Comment by jmcarcell** [comment]
Ready to be merged and tested!

**Comment by jmcarcell** [comment]
@spackbot fix style

**Comment by jmcarcell** [comment]
Can this be merged? We have been using this for a while in our nightly builds
@alalazo

**Comment by jmcarcell** [comment]
This is ready to be merged!, @vvolkl can you approve please?

**Review by vvolkl — APPROVED** [review]

**Review by alalazo — APPROVED** [review]
Not really a fan of `LD_LIBRARY_PATH` being needed, so if you can avoid that, please submit another PR :slightly_smiling_face:

**Comment by sethrj** [comment]
Agree with @alalazo here. @pcanal Does ROOT perhaps offer something like a  `ROOTPATH` environment variable to be used for dictionaries? If this is the only way, we may want to do something similar for Celeritas too 🙄

**Comment by pcanal** [comment]
@sethrj 

TLDR: yes `ROOT_LIBRARY_PATH`

The need for `LD_LIBRARY_PATH` is only for libraries (including libraries containing only a ROOT dictionary) that needs to be autoloaded.  If a library is hard linked and found via the rpath, ROOT should still find the required auxiliary files without using the `LD_LIBRARY_PATH`  (assuming those files are located in the same file system directories as the library).

For autoloading libraries, by definition, we need a source of information that is external to the executable.
By default ROOT use the list of directories provided by:
```
environment variable: ROOT_LIBRARY_PATH
ROOT resource (eg in .rootrc): Root.DynamicPath
On Linux and MacOs environment variable: LD_LIBRARY_PATH
On MacOS, environment variable: DYLD_LIBRARY_PATH
On Windows, environment vairable: PATH
System search path
```
